web/scripts/main-index.js

```diff
@@ -127,20 +127,35 @@ function makeDataProvider() {
     };
 }
 
+const LB_CLASSES = [
+    // Traditional
+    "-97", "-106", "-115", "-123", "-132", "-149", "-165", "-182", "-198", "198+", "-220", "-243", "-276", "-309", "309+",
+    // IPF Men
+    "-117", "-130", "-146", "-163", "-183", "-205", "-231", "-265", "265+",
+    // IPF Women
+    "-94.75", "-104", "-115", "-126", "-139", "-158", "-185", "185+"
+]
+
+const KG_CLASSES = [
+    // Traditional
+    "-44", "-48", "-52", "-56", "-60", "-67.5", "-75", "-82.5", "-90", "90+", "-100", "-110", "-125", "-140", "140+",
+    // IPF Men
+    "-53", "-59", "-66", "-74", "-83", "-93", "-105", "-120", "120+",
+    // IPF Women
+    "-43", "-47", "-52", "-57", "-63", "-72", "-84", "84+"
+]
+
 function generateWeightClasses() {
-    for(var i = 0; i < selClass.options.length; i++) {
-        var option = selClass.options[i];
-        var num = option.text.replace(/[^0-9\.]/g, '');
-        console.log(num);
-        if(option.text.indexOf('lbs') >= 0 && selWeightType.value === "kg") {
-            option.text = option.text.replace(num, Math.round(common.lbs2kg(num)));
-            option.text = option.text.replace('lbs', 'kg');
+    if(selWeightType.value === "lb") {
+        for(var i = 1; i < selClass.options.length; i++) {
```

Comment:
Should this be `<=`? Also below. If `selClass.options.length == 1` then the loop will never execute.

Instead of duplicating the loop, it might be better to just use `selWeightType` to set a local variable, for example `var weightclasses = (selWeightType.value === "lb" ? LB_CLASSES : KG_CLASSES);`. Then there only has to be one loop, which is a little nicer.